src/StreamSnapshotProjection.php

```diff
@@ -31,15 +31,16 @@ public function __construct(ReadModelProjector $readModelProjector, string $stre
     {
         $this->readModelProjector = $readModelProjector;
         $this->streamName = $streamName;
-    }
 
-    public function __invoke(bool $keepRunning = true)
-    {
         $this->readModelProjector
             ->fromStream($this->streamName)
             ->whenAny(function ($state, Message $event): void {
                 $this->readModel()->stack('replay', $event);
-            })
```

Comment:
Any special reason why moving this logic into the ctor?

There's no need to store the `streamName` if you do so.

You could then do:

```php
  1 <?php                                                                                                                   
  2                                                                                                                         
  3 public function __construct(ReadModelProjector $readModelProjector, string $streamName)                                                                                                                                                   
  4 {                                                                                                                       
  5     $this->readModelProjector = $readModelProjector                                                                     
  6         ->fromStream($streamName)                                                                                       
  7         ->whenAny(function ($state, Message $event): void {                                                             
  8             $this->readModel()->stack('replay', $event);                                                                
  9         });                                                                                                             
 10 }         
```